### PR TITLE
parametric: support otel config tests in ruby

### PR DESCRIPTION
**Pull request by mabdinur**
## Motivation

Enables `Test_Otel_Env_Vars` for ruby tracers.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/build/docker/ruby/parametric/server.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@
 OpenTelemetry::SDK.configure # Initialize OpenTelemetry
 
 Datadog.configure do |c|
-  c.diagnostics.debug = true # When tests fail, ensure there's enough data to debug the failure.
+  # c.diagnostics.debug = true # When tests fail, ensure there's enough data to debug the failure.
```
Comment (mabdinur) [inline utils/build/docker/ruby/parametric/server.rb:32]: Enabling debug logging here interferes with the test. We have to find a workaround for this. Maybe update the test cases?

Comment (mabdinur) [inline utils/build/docker/ruby/parametric/server.rb:32]:
```suggestion
  if ENV['DD_TRACE_DEBUG'].nil?
    # When tests fail, ensure there's enough data to debug the failure.
    # If DD_TRACE_DEBUG is set do not override this configuration.
    c.diagnostics.debug = true
  end
```

File: tests/parametric/test_otel_env_vars.py [hunk 2]
```diff
@@ -160,6 +160,7 @@ def test_otel_log_level_to_debug_mapping(self, test_agent, test_library):
         assert (resp["dd_log_level"] == "debug") or (resp["dd_log_level"] is None)
 
     @missing_feature(context.library == "nodejs", reason="this setting is not exposed in the Node.js config object")
+    @missing_feature(context.library == "ruby", reason="does not support enabling opentelemetry via DD_TRACE_OTEL_ENABLED")
```
Comment (mabdinur) [inline tests/parametric/test_otel_env_vars.py:163]:
```suggestion
    @missing_feature(
        context.library == "ruby", reason="does not support enabling opentelemetry via DD_TRACE_OTEL_ENABLED"
    )
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Error in parametric runs in CI

**Comment by mabdinur** [comment]
> Error in parametric runs in CI

I generated the protos using an incompatible version of grpcio-tools. I will re-generate the protos with the correct version.

**Comment by mabdinur** [comment]
The failing ruby parametric tests are unrelated and will be resolved in a future PR: [Details](https://github.com/DataDog/system-tests/actions/runs/9453083596/job/26064276490?pr=2538)

File: tests/parametric/test_otel_env_vars.py [hunk 3]
```diff
@@ -160,13 +161,17 @@ def test_otel_log_level_to_debug_mapping(self, test_agent, test_library):
...
+        context.library == "ruby", reason="does not support enabling opentelemetry via DD_TRACE_OTEL_ENABLED"
+    )
     @pytest.mark.parametrize("library_env", [{"DD_TRACE_OTEL_ENABLED": "true", "OTEL_SDK_DISABLED": "true"}])
     def test_dd_trace_otel_enabled_takes_precedence(self, test_agent, test_library):
         with test_library as t:
             resp = t.get_tracer_config()
         assert resp["dd_trace_otel_enabled"] == "true"
 
     @missing_feature(context.library == "nodejs", reason="this setting is not exposed in the Node.js config object")
+    @missing_feature(context.library == "ruby", reason="does not support enabling opentelemetry via DD_TRACE_OTEL_ENABLED")
```
Comment (mabdinur) [inline tests/parametric/test_otel_env_vars.py:174]:
```suggestion
    @missing_feature(
        context.library == "ruby", reason="does not support enabling opentelemetry via DD_TRACE_OTEL_ENABLED"
    )
```

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**